src/app/components/sendForm/useClearFormOnAccountSwitch.ts

```diff
@@ -0,0 +1,21 @@
+import useWalletSelector from '@hooks/useWalletSelector';
+import { useEffect, useRef, useState } from 'react';
+
+export default function useClearFormOnAccountSwitch() {
+  const { selectedAccount } = useWalletSelector();
+  const [isAccountSwitched, setIsAccountSwitched] = useState(false);
+
+  const isFirstRender = useRef(false);
+  useEffect(() => {
+    if (isFirstRender.current) {
+      setIsAccountSwitched(true);
+    } else {
+      //if first render then dont clear form input fields
+      isFirstRender.current = true;
+    }
+  }, [selectedAccount]);
```

Comment:
@Imamah-Zafar nice job with the hook, I like the abstraction.

just wondering, did you test this with multiple account switches? after isAccountSwitched is set to true once, it looks like it won't trigger again

Reply:
yes you're right. It wasn't being triggered for multiple account switches, I updated the dependency array for the useEffect so it would get called when the account changes . It's working now for multiple switches 



https://github.com/secretkeylabs/xverse-web-extension/assets/88320460/02ba60fb-4fec-4d83-9d3d-faeee03823a5